src/App.tsx

```diff
@@ -3,91 +3,111 @@ import './App.css';
 import {Settings} from "./assets/settings/Settings";
 import {Counter} from "./assets/counter/Counter";
 
+const instructionsMessage = 'Enter values and press \'set\'';
+const errorMessage = 'Incorrect value!';
+
+export type InitialValueType = {
+    maxValue: number,
+    startValue: number
+};
+
+export type ControlInfoType = {
+    isError: boolean
+    message: string
+    isVisible: boolean
+};
+
+export type ControlVisibleButtonType = {
+    isResultMax: boolean
+    isInitialValueSet: boolean
+};
+
 function App() {
-    const [maxValue, setMaxValue] = useState('5');
-    const [startValue, setStartValue] = useState('0');
-    const [count, setCount] = useState('Enter values and press \'set\'');
-    const [error, setError] = useState(false);
-    const [disabled, setDisabled] = useState(false);
-    const [settingDisabled, setSettingDisabled] = useState(false);
+
+    const [initialValue, setInitialValue] = useState<InitialValueType>({
+        maxValue: 1,
+        startValue: 0
+    });
+    const [resultValue, setResultValue] = useState(0);
+    const [controlInfo, setControlInfo] = useState<ControlInfoType>({
+        isError: false,
+        message: instructionsMessage,
+        isVisible: true
+    });
+    const [controlVisibleButton, setControlVisibleButton] = useState<ControlVisibleButtonType>({
+        isResultMax: false,
+        isInitialValueSet: false
+    });
 
     useEffect(() => {
-        let max = localStorage.getItem('max')
-        let start = localStorage.getItem('start')
+        const max =  localStorage.getItem('max');
+        const start = localStorage.getItem('start');
         if (max && start) {
-            setMaxValue(max)
-            setStartValue(start)
-            setCount(start)
+            let maxLocalValue = JSON.parse(max);
+            let startLocalValue = JSON.parse(start);
+            setInitialValue({maxValue: maxLocalValue, startValue: startLocalValue})
+            setResultValue(initialValue.startValue)
         }
-    }, [])
+    }, []);
 
     const counterIncrement = () => {
-        if (+count === +maxValue - 1) {
-            setDisabled(true)
-            return setCount(`${+count +1}`)
-        } else {
-            return setCount(`${+count +1}`)
+        setResultValue(resultValue + 1);
+        if (resultValue === initialValue.maxValue - 1 ) {
+            setControlVisibleButton({...controlVisibleButton, isResultMax: true})
```

Comment:
if I remember well you can write like: 
```suggestion
            setControlVisibleButton({ isResultMax: true})
```

Reply:
If we write like this, we will get an error
https://prnt.sc/74QwcIKnhL44